actioncable.go

```diff
@@ -94,6 +97,44 @@ func (c *Client) Subscribe(channel string) (<-chan *EventOrErr, error) {
 	}
 
 	c.subscriptions[channel] = make(chan *EventOrErr, 32)
+	c.subscData[channel], err = json.Marshal(&innerIdentifier{Channel: channel})
+	if err != nil {
+		panic(err)
+	}
+
+	// Add subscription command to outbound queue.
+	go func() {
+		c.subc <- channel
+	}()
+
+	return c.subscriptions[channel], nil
+}
+
+var ErrNoChannelName = errors.New("Subscription data must contain a channel name")
+
+// SubscribeWith subscribes to a channel giving parameters
+// the params must contain a field "channel" containing the channe to subscribe to
+func (c *Client) SubscribeWith(params json.RawMessage) (<-chan *EventOrErr, error) {
```

Comment:
I didn't chose `map[string]interface{}` although I considered it. This is mainly because it let me use a marshalled struct for the params. I have something that looks like:

``
type ConnectionData struct {
    Channel string `json:"channel"`
    SubscribeTo []string `jsob:"subscribe_to"` // Passed to the rails code
}
```

The other solution would have been to pass an `interface{}` that would have been marshalled within the function. Perhaps it would be possible to marshall it and override the `channel` JSON attribute doing so. Might be complicated though.